diff --git a/src/serialport/qserialport_unix.cpp b/src/serialport/qserialport_unix.cpp
index 7e642e5..06346e4 100644
--- a/src/serialport/qserialport_unix.cpp
+++ b/src/serialport/qserialport_unix.cpp
@@ -32,13 +32,13 @@
 
 #ifdef Q_OS_LINUX
 
-# ifdef Q_OS_ANDROID
+# ifdef __ANDROID__
 #  include <android/api-level.h>
 # else
 #  define __ANDROID_API__ 16
 # endif
 
-# if !defined(Q_OS_ANDROID) || (!defined(Q_PROCESSOR_X86) && __ANDROID_API__ < 16)
+# if !defined(__ANDROID__) || (!defined(Q_PROCESSOR_X86) && __ANDROID_API__ < 16)
 struct termios2 {
     tcflag_t c_iflag;       /* input mode flags */
     tcflag_t c_oflag;       /* output mode flags */
@@ -78,7 +78,7 @@ QString serialPortLockFilePath(const QString &portName)
         << QStringLiteral("/var/tmp")
         << QStringLiteral("/var/lock/lockdev")
         << QStringLiteral("/run/lock")
-#ifdef Q_OS_ANDROID
+#ifdef __ANDROID__
         << QStringLiteral("/data/local/tmp")
 #endif
         << QStandardPaths::writableLocation(QStandardPaths::TempLocation);
