diff --git a/configure.ac b/configure.ac
index 453ff1c56..30b35081a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -920,7 +920,7 @@ solaris*)
     fi
     ;;
 
-linux-gnu*|k*bsd*-gnu*|linux-musl*)
+linux-gnu*|k*bsd*-gnu*|linux-musl*|linux-android*)
     using_freetype_fontconfig=yes
     using_x11=yes
     build_skia=yes
@@ -1093,28 +1093,6 @@ dragonfly*)
     _os=DragonFly
     ;;
 
-linux-android*)
-    # API exists, but seems not really usable since Android 7 AFAIK
-    usable_dlapi=no
-    using_freetype_fontconfig=yes
-    using_headless_plugin=no
-    using_x11=no
-    build_crypto=no
-    test_openldap=no
-    test_system_freetype=no
-    test_webdav=no
-    disable_database_connectivity_dependencies
-    enable_lotuswordpro=no
-    enable_mpl_subset=yes
-    enable_cairo_canvas=no
-    enable_coinmp=yes
-    enable_lpsolve=no
-    enable_odk=no
-    enable_python=no
-    enable_xmlhelp=no
-    _os=Android
-    ;;
-
 haiku*)
     using_freetype_fontconfig=yes
     using_x11=no
