diff --git a/common/wscript b/common/wscript
index 178d4c1..db537e1 100644
--- a/common/wscript
+++ b/common/wscript
@@ -30,7 +30,7 @@ def create_jack_process_obj(bld, target, sources, uselib=None, framework=None):
             process.framework = framework
         env_includes = ['../macosx', '../posix', '../macosx/coreaudio']
     if bld.env['IS_LINUX']:
-        env_includes = ['../linux', '../posix', '../linux/alsa']
+        env_includes = ['../linux', '../posix', '../linux/alsa', '../android']
     if bld.env['IS_FREEBSD']:
         env_includes = ['../freebsd', '../posix', '../solaris/oss']
     if bld.env['IS_SUN']:
@@ -87,14 +87,14 @@ def build(bld):
             'JackDebugClient.cpp',
             'timestamps.c',
             'promiscuous.c',
-            '../posix/JackPosixThread.cpp',
+            '../android/JackAndroidThread.cpp',
             '../posix/JackPosixProcessSync.cpp',
             '../posix/JackPosixMutex.cpp',
             '../posix/JackSocket.cpp',
             '../linux/JackLinuxFutex.cpp',
             '../linux/JackLinuxTime.c',
             ]
-        includes = ['../linux', '../posix'] + includes
+        includes = ['../linux', '../posix', '../android'] + includes
         uselib.append('RT')
         uselib.append('DL')
 
diff --git a/posix/JackPosixThread.h b/posix/JackPosixThread.h
index 599bf49..b4a57fa 100644
--- a/posix/JackPosixThread.h
+++ b/posix/JackPosixThread.h
@@ -21,6 +21,13 @@ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 #ifndef __JackPosixThread__
 #define __JackPosixThread__
 
+#ifdef __ANDROID__
+
+#include "JackAndroidThread.h"
+typedef Jack::JackAndroidThread Jack::JackPosixThread
+
+#else
+
 #include "JackThread.h"
 #include <pthread.h>
 
@@ -85,5 +92,6 @@ SERVER_EXPORT void ThreadExit();
 
 } // end of namespace
 
+#endif
 
 #endif
